cmd/genji/shell/command.go

```diff
@@ -136,3 +140,132 @@ func displaySuggestions(in string) error {
 	fmt.Println()
 	return nil
 }
+
+// dumpTable display the content of the given table as SQL statements.
+func dumpTable(db *genji.DB, t *database.Table) error {
+	// Get table info.
+	ti, err := t.Info()
+	if err != nil {
+		return err
+	}
+
+	fields := ti.FieldConstraints
+	if len(fields) == 0 {
+		fmt.Printf("CREATE TABLE %s;\n", t.Name())
+	} else {
+		fmt.Printf("CREATE TABLE %s(", t.Name())
+		for i, f := range fields {
+			fmt.Printf("%s %s", fields[i].Path, strings.ToUpper(fields[i].Type.String()))
+			if f.IsPrimaryKey {
+				fmt.Printf(" PRIMARY KEY")
+			}
+
+			if f.IsNotNull {
+				fmt.Printf(" NOT NULL")
+			}
+
+			// Don't print the last comma.
+			if i == (len(fields) - 1) {
+				fmt.Println()
+				break
+			}
+
+			fmt.Println(",")
+		}
+
+		fmt.Println(");")
+	}
+
+	// Indexes section.
+	indexes, err := t.Indexes()
+	if err != nil {
+		return err
+	}
+
+	// Display indexes.
+	for _, index := range indexes {
+		u := ""
+		if index.Opts.Unique {
+			u = " UNIQUE"
+		}
+
+		info := fmt.Sprintf("CREATE%s INDEX %s ON %s (%s);", u, index.Opts.IndexName, index.Opts.TableName, index.Opts.Path)
+		fmt.Println(info)
+	}
+
+	// Inserts section.
+	return db.View(func(tx *genji.Tx) error {
+		q := fmt.Sprintf("SELECT * FROM %s", t.Name())
+		res, err := tx.Query(q)
+		if err != nil {
+			return err
+		}
+		defer res.Close()
+
+		c, err := res.Count()
+		if err != nil {
+			return err
+		}
+		// Check if the table is empty.
+		if c == 0 {
+			return nil
+		}
+
+		// Display inserts.
+		return res.Iterate(func(d document.Document) error {
+			var buf bytes.Buffer
+			err := document.ToJSON(&buf, d)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("INSERT INTO %s VALUES %s;\n", t.Name(), buf.String())
+			return nil
+		})
+
+	})
+}
+
+func runDumpCmd(db *genji.DB, table string) error {
+	fmt.Println("BEGIN TRANSACTION;")
+	defer fmt.Println("COMMIT;")
```

Comment:
If there is an error in the middle of the dump, we will output `COMMIT` anyway, we should not do this. Let's print it when it's supposed to be printed instead.

Reply:
If there is an error what kind of message? sqlite3 print `ROLLBACK; --due to an error` should we do it or don't print anything. 
```SQL
BEGIN TRANSACTION;
error: XXX 
```

Reply:
I think we should print the rollback yes, this might be streamed to another instance of Genji who will get statements one by one, we want to make sure that if something happens in the middle of the process, we keep a consistent state.